include/mbedtls/check_config.h

```diff
@@ -1021,6 +1025,11 @@
 #error "MBEDTLS_X509_CSR_WRITE_C defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_X509_TRUSTED_CERTIFICATE_CALLBACK) && \
+            ( !defined(MBEDTLS_X509_CSR_WRITE_C) )
```

Comment:
That doesn't feel correct to me: after re-reading the documentation for that option I don't see why it should depend on the library's ability to write CSRs. I'd be much less surprised by a dependency on `X509_CRT_PARSE_C`. What prompted you to add this dependency?

Reply:
I apologize for this mistake, but rechecking the code I think you're right:
- [ssl.h](https://github.com/Mbed-TLS/mbedtls/blob/262851df1cc4f090b3e2920cd647d5e22c9bb2f1/include/mbedtls/ssl.h#L1458)
- [ssl.h](https://github.com/Mbed-TLS/mbedtls/blob/262851df1cc4f090b3e2920cd647d5e22c9bb2f1/include/mbedtls/ssl.h#L3298)
- [x509_crt.h](https://github.com/valeriosetti/mbedtls/blob/7cd1ebe0bb739866124dd376f53711bc900ed6ca/include/mbedtls/x509_crt.h#L847)
I don't remember how I picked that dependency, but at this point I agree that it was wrong

Reply:
Well no need to apologize, mistakes happen, that's one of the reasons we have reviews! As long as we can discuss and eventually come to an agreement on what's correct, no big trouble.